src/gui/qgstemporallayerwidget.h

```diff
@@ -0,0 +1,134 @@
+/***************************************************************************
+                         qgstemporallayerwidget.h
+                         ------------------------------
+    begin                : January 2020
+    copyright            : (C) 2020 by Samweli Mwakisambwe
+    email                : samweli at kartoza dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSTEMPORALLAYERWIDGET_H
+#define QGSTEMPORALLAYERWIDGET_H
+
+#include "ui_qgstemporallayerwidgetbase.h"
+#include "qgis_gui.h"
+#include "qgsmapcanvas.h"
+#include "qgsrasterlayer.h"
+
+
+class QgsMapLayer;
+
+/**
+ * \ingroup gui
+ * \class QgsTemporalLayerWidget
+ * A widget for filtering temporal layer based on its available time values
```

Comment:
```suggestion
 * A widget for configuring the temporal properties for a raster layer.
```

Reply:
Better to keep this one specific to rasters -- mesh and vector will have very different widget requirements